include/jemalloc/internal/mutex_types.h

```diff
@@ -7,25 +7,25 @@ typedef struct malloc_mutex_s malloc_mutex_t;
 #  define MALLOC_MUTEX_INITIALIZER
 #elif (defined(JEMALLOC_OS_UNFAIR_LOCK))
 #  define MALLOC_MUTEX_INITIALIZER					\
-     {OS_UNFAIR_LOCK_INIT, WITNESS_INITIALIZER("mutex", WITNESS_RANK_OMIT)}
+     {{{OS_UNFAIR_LOCK_INIT}}, WITNESS_INITIALIZER("mutex", WITNESS_RANK_OMIT)}
```

Comment:
I did some reading (of a C11 draft) and found that initialization happens in declaration order unless designated initializers are used.  So, if we're going to use this union trick, I think we need to put the witness structure first so that its initialization is overwritten by the subsequent fields.